vcd/datasource_vcd_nsxt_alb_pool.go

```diff
@@ -0,0 +1,253 @@
+package vcd
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func datasourceVcdAlbPool() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: datasourceVcdAlbPoolRead,
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"edge_gateway_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway ID in which ALB Pool should be created",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of ALB Pool",
+			},
+			"enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Defines if NSX-T ALB Pool is enabled or disabled",
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Description of ALB Pool",
+			},
+			"algorithm": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Algorithm for choosing pool members (default LEAST_CONNECTIONS)",
+			},
+
+			"default_port": &schema.Schema{
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Default Port defines destination server port used by the traffic sent to the member (default 80)",
+			},
+			"graceful_timeout_period": &schema.Schema{
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Maximum time in minutes to gracefully disable pool member (default 1)",
+			},
+			"member": {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Description: "ALB Pool Members",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": &schema.Schema{
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "Shows is the member is enabled or not",
+						},
+						"ip_address": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "IP Address of pool member",
+						},
+						"port": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "Service port",
+						},
+						"ratio": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "Load ratio",
+						},
+						"marked_down_by": {
+							Type:        schema.TypeSet,
+							Computed:    true,
+							Description: "Marked down by",
```

Comment:
Better add more in-depth description from the docs.

Reply:
Changed in both resource and data source to 
```go
Description: "Marked down by provides a set of health monitors that marked the service down",
```